libs/designer/src/lib/ui/Designer.tsx

```diff
@@ -108,16 +109,20 @@ export const CanvasFinder = () => {
   return null;
 };
 
+export const SearchBrowsePreloader = () => {
+  usePreloadOperationsQuery();
+  usePreloadConnectorsQuery();
+  return null;
+};
+
 export const Designer = (props: DesignerProps) => {
   const { backgroundProps } = props;
 
   const [nodes, edges, flowSize] = useLayout();
   const isEmpty = useIsGraphEmpty();
   const isReadOnly = useReadOnly();
   const dispatch = useDispatch();
-
-  usePreloadOperationsQuery();
-  usePreloadConnectorsQuery();
+  const [initializationStage, setInitializationStage] = useState<'Start' | 'Loading' | 'Finished'>('Start');
```

Comment:
Future note: This might be worth tracking in a way as well that portal can use it or redux can use it to start tracking dirty state.